DNN Platform/Library/Services/Log/EventLog/LogProperties.cs

```diff
@@ -25,8 +29,52 @@ public string Summary
 
                 return summary;
             }
-        }
-
+        }
+
+        /// <inheritdoc />
+        ILogDetailInfo IList<ILogDetailInfo>.this[int index]
+        {
+            get => (ILogDetailInfo)this[index];
+            set => this[index] = value;
+        }
+
+        /// <inheritdoc />
+        public int IndexOf(ILogDetailInfo item) =>
+            base.IndexOf(item);
+
+        /// <inheritdoc />
+        public void Insert(int index, ILogDetailInfo item) =>
+            base.Insert(index, item);
+
+        /// <inheritdoc />
+        public void Add(ILogDetailInfo item) => 
+            base.Add(item);
+
+        /// <inheritdoc />
+        public bool Contains(ILogDetailInfo item) => 
+            base.Contains(item);
+
+        /// <inheritdoc />
+        public void CopyTo(ILogDetailInfo[] array, int arrayIndex) => 
+            base.CopyTo(array, arrayIndex);
+
+        /// <inheritdoc />
+        IEnumerator<ILogDetailInfo> IEnumerable<ILogDetailInfo>.GetEnumerator() => 
+            this.ToArray().Cast<ILogDetailInfo>().GetEnumerator();
+
+        /// <inheritdoc />
+        public bool Remove(ILogDetailInfo item)
+        {
+            var index = base.IndexOf(item);
+            if (index >= 0)
+            {
+                this.RemoveAt(index);
+                return true;
+            }
+
+            return false;
+        }

```

Comment:
@bdukes I updated this code in the latest push to use `IndexOf` instead of `Contains` because it uses the native `Array.IndexOf` implementation which will be slightly faster. 

In this code snippet we will iterate through all of the items in the collection once O(n) vs the other approach we would iterate through everything twice O(2n). Granted the big-O simplifies down to O(n) for both scenarios, but this should be slightly faster.